frontend/src/tests/assignment-tests.js

```diff
@@ -1,10 +1,125 @@
-import { it } from "./utils";
+import PropTypes from "prop-types";
+import {
+    it,
+    expect,
+    beforeAll,
+    checkPropTypes,
+    assignmentPropTypes
+} from "./utils";
+import { databaseSeeder } from "./setup";
 
-// TODO: Remove eslint disable
-// eslint-disable-next-line
 export function assignmentsTests({ apiGET, apiPOST }) {
-    it.todo("get assignments for session");
-    it.todo("get assignments for position");
-    it.todo("create and delete assignment for position");
-    it.todo("update assignment dates/note/offer_override_pdf");
+    const session = databaseSeeder.seededData.session;
+    const position = databaseSeeder.seededData.position;
+    const assignment = databaseSeeder.seededData.assignment;
+
+    beforeAll(async () => {
+        await apiPOST("/debug/restore_snapshot");
+    }, 30000);
+
+    it("get assignments for session", async () => {
+        const resp = await apiGET(`/admin/sessions/${session.id}/assignments`);
+
+        expect(resp).toMatchObject({ status: "success" });
+        // make sure we got the seeded assignment data
+        expect(resp.payload).toMatchObject([assignment]);
+        // check the type of payload
+        checkPropTypes(PropTypes.arrayOf(assignmentPropTypes), resp.payload);
+    });
+
+    it("update assignment dates/note/offer_override_pdf/hours", async () => {
+        // update dates/note/offer_override_pdf/hours"
+        const updatedAssignmentData = {
+            ...assignment,
+            start_date: "2019-09-05T00:00:00.000Z",
+            end_date: "2019-12-25T00:00:00.000Z",
+            note: "updated",
+            offer_override_pdf: "pdf",
+            hours: 80
+        };
+
+        const resp = await apiPOST(`/admin/assignments`, updatedAssignmentData);
+        expect(resp).toMatchObject({ status: "success" });
+
+        const { payload: updatedAssignment } = resp;
+        // make sure the propTypes are right
+        checkPropTypes(assignmentPropTypes, updatedAssignment);
+        expect(updatedAssignment.id).not.toBeNull();
+        expect(updatedAssignment.id).toEqual(assignment.id);
+
+        // fetch all assignments and make sure the newly added
+        // assignment data is there
+        const { payload: withUpdatedAssignment } = await apiGET(
+            `/admin/sessions/${session.id}/assignments`
+        );
+
+        expect(withUpdatedAssignment.map(x => x.id)).toContain(
+            updatedAssignment.id
+        );
+        expect(
+            withUpdatedAssignment.filter(s => s.id === updatedAssignment.id)
+        ).toContainObject(updatedAssignment);
+
+        expect(withUpdatedAssignment.length).toEqual(1);
+    });
+
```

Comment:
maybe we can have some error checking later when the backend is fully functional?

Reply:
Maybe we can do it in another PR? I was just finishing all the test cases that was originally here. Needs to investigate for error checking.